include/lsst/meas/base/FluxUtilities.h

```diff
@@ -209,6 +209,21 @@ class FluxTransform : public BaseTransform {
     MagResultKey _magKey;
 };
 
+/**
+ *  Temporarily replace negative fluxes with NaNs
+ *
+ *  Instantiating a NoThrowOnNegativeFluxContext object will cause afw::image::Calib
+ *  objects to return NaN, rather than throwing, when asked to convert a negative flux to
+ *  a magnitude for the lifetime of the NoThrowOnNegativeFluxContext.
+ */
+class NoThrowOnNegativeFluxContext {
+public:
+    NoThrowOnNegativeFluxContext();
+    ~NoThrowOnNegativeFluxContext();
+private:
+    bool _throwOnNegative;
+};
+
```

Comment:
I like this a lot.  I'd like it even better if we could integrate it more closely with Calib, but I don't think that's appropriate for this ticket (it's more disruptive), so I've opened DM-3489 for that.